### PR TITLE
Automate dependency checks with dependabot

### PR DESCRIPTION
**Pull request by bishal7679**
Fixes :- #1274

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
+updates:
+- package-ecosystem: maven
```
Comment (andrewsomething) [inline .github/dependabot.yml:3]:
```suggestion
- package-ecosystem: gomod
```

`doctl` is a Go project that manages its dependencies using `gomod`.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  directory: "/"
```
Comment (andrewsomething) [inline .github/dependabot.yml:4]:
```suggestion
  directory: "/"
  vendor: true
```

We vendor our dependencies in the `vendor/` directory.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  schedule:
+    interval: weekly
+  open-pull-requests-limit: 10
+  labels:
+  - skip-changelog
```
Comment (andrewsomething) [inline .github/dependabot.yml:9]: I think we can remove this. We want to track all changes in the changelog.

**Review by andrewsomething — CHANGES_REQUESTED** [review]
Hi @bishal7679,

Thanks for the PR! Generally we've been pretty conservative about update dependencies only as needed in this project, but enabling dependabot might be a good experiment.

`doctl`  is a Go project. I've suggested some updates to reflect that.

**Comment by bishal7679** [comment]
> Hi @bishal7679,
> 
> Thanks for the PR! Generally we've been pretty conservative about update dependencies only as needed in the is project, but enabling dependabot might be a good experiment.
> 
> `doctl` is a Go project. I've suggested some updates to reflect that.

Oh! my bad. I have mistakenly used maven its for java project.
Nice suggestion!

**Comment by bishal7679** [comment]
@andrewsomething pls Take a look again!

**Review by andrewsomething — APPROVED** [review]
:+1:  Looks good! Thank you for the contribution.

**Pull request merged**